### ISSUE TITLE
Crash of the python kernel in jupyter

### ISSUE THREAD

**Issue by igudich** [mantidproject/dataset#116]
```
N = 6
M = 4
d1 = Dataset()
d1[Coord.X] = ([Dim.X], np.arange(N+1).astype(np.float64))
d1[Coord.Y] = ([Dim.Y], np.arange(M+1).astype(np.float64))

arr1 = np.arange(N*M).reshape(N,M).astype(np.float64)
#TODO copy not needed after correct treatment of strides
arr2 = np.transpose(arr1).copy();

d1[Data.Value, "A"] = ([Dim.X, Dim.Y], arr1)
d1[Data.Value, "B"] = ([Dim.Y, Dim.X], arr2)
d1[Dim.X]  # this line crashed the python kernel
```

**Comment by nvaytet** [comment]
Is this because you need to have `d1[Dim.X, :]` to slice the `Dataset`?

**Comment by nvaytet** [comment]
So it turns out the `.def("__getitem__", [](Dataset &self, const gsl::index index)` is being called here. We need to understand why a conversion from `Dim.X` to `gsl::index` is being made.

**Comment by nvaytet** [comment]
Probably because internally Dim.X is just an integer?

**Comment by SimonHeybrock** [comment]
It is a strongly typed enum. Documentation is here: https://pybind11.readthedocs.io/en/stable/classes.html#enumerations-and-internal-types. We are doing things correctly as far as I can see (e.g., we do not call `export_values()`)?

**Comment by nvaytet** [comment]
A workaround would be to just have a __getitem__ that takes a Dim as an argument and throws an error, but this would only cover that case. It should fail in the same way as if you supply just a string, with a type error message. Doesn’t feel super nice to have a getitem that just throws an error

**Comment by nvaytet** [comment]
This was fixed in #128
